<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>中断promise链</title>
</head>
<body>
  <script>

    /* 
    中断promise链的方法：有且只有一个，就是必须返回一个pending状态的promise对象
    */

    let p = new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve('ok');
        // rejected('error');
      }, 1000);
    });

    p.then(value => {
      console.log(111);
      // 有且只有一种方法
      return new Promise(() => {}); //输出结果只有 111 ，在此之后被中断
    }).then(value => {
      console.log(222);
    }).then(value => {
      console.log(333);
    }).catch(reason => {
      console.log(reason); // 在最后的位置使用catch的方法来捕获异常就可以处理，不需要在上面的then中进行
    })
  </script>
</body>
</html>